### PR TITLE
Add support for IEEE 754 NaN / infinity

### PR DESCRIPTION
**Pull request by CandyGumdrop**
This enables unpacking of 32-bit and 64-bit floating-point NaN, +infinity and -infinity, to the values `:NaN`, `:inf` and `:"-inf"`, but does not support packing those atoms back into Messagepack floats (as these atoms would otherwise be packed into strings and this could be considered a breaking change).

I chose these atom values as there is a precedent in the [mongodb library](https://github.com/ankhers/mongodb), but these could perhaps be changed to other values, such as `:nan` rather than `:NaN`.

None of this is really ideal since Erlang doesn't support NaN or infinity in floats, but this at least provides some handling of these values which are supported by Messagepack.

### CODE REVIEW

**Comment by lukaszsamson** [comment]
I just got bitten by this. Can we get this merged?
If there is no plan to support NaN and Inf at least msgpax could raise a more meaningful error message than
```
** (Msgpax.UnpackError) invalid format, first byte: 203
    (msgpax 2.2.4) lib/msgpax.ex:252: Msgpax.unpack!/2
```

First byte 203 is definitely a valid format ;)

File: test/msgpax_test.exs [hunk 1]
```diff
@@ -104,8 +104,28 @@ defmodule MsgpaxTest do
     assert_format Atom, <<171>>, "Elixir.Atom"
   end
 
-  test "float" do
+  test "floats" do
     assert_format 42.1, <<203>>
+
+    for {packed, value} <- %{
+          # 32-bit
```
Comment (qcam) [inline test/msgpax_test.exs:111]:
```suggestion
          # 32-bit.
```

File: test/msgpax_test.exs [hunk 2]
```diff
@@ -104,8 +104,28 @@ defmodule MsgpaxTest do
...
+          <<202, 0x7FC00000::32>> => Msgpax.NaN,
+          <<202, 0x7F800000::32>> => Msgpax.Infinity,
+          <<202, 0xFF800000::32>> => Msgpax.NegInfinity,
+          # 64-bit
```
Comment (qcam) [inline test/msgpax_test.exs:115]:
```suggestion
          # 64-bit.
```

File: lib/msgpax/packer.ex [hunk 3]
```diff
@@ -103,6 +103,14 @@ defprotocol Msgpax.Packer do
   It returns an iodata result.
   """
   def pack(term)
+
+  Kernel.def(pack_nan(), do: <<0xCB, -1::64>>)
```
Comment (qcam) [inline lib/msgpax/packer.ex:107]: Shall we write some docs for these 2 packers as well?

Comment (lexmag) [reply lib/msgpax/packer.ex:107]: Yes. Actually, I'll move packing addition to another PR.

**Review by lexmag — APPROVED** [review]

**Review by qcam — APPROVED** [review]

**Comment by lexmag** [comment]
@CandyGumdrop thank you!

**Pull request merged**